react-native/android/app/src/main/java/io/keybase/android/modules/KeybaseEngine.java

```diff
@@ -31,17 +33,17 @@ public ReadFromKBLib(ReactApplicationContext reactContext) {
 
         @Override
         public void run() {
-            // TODO: There may be a race condition here...
-            // It will fail if you try to run .getJSModule
-            // before react has loaded.
-
-            while (!Thread.currentThread().isInterrupted()) {
+            do {
                 final String data = ReadB64();
 
+                if (!reactContext.hasActiveCatalystInstance()) {
```

Comment:
Potential for dropped messages here, but in my testing only results in an empty string since we call `Reset()` after destroying this.